pkg/operator/ceph/cluster/cluster.go

```diff
@@ -460,6 +460,22 @@ func (c *cluster) preMonStartupActions(cephVersion cephver.CephVersion) error {
 // It gets executed right after the main mon Start() method
 // Basically, it is executed between the monitors and the manager sequence
 func (c *cluster) postMonStartupActions() error {
+	// rotate mon cephx keys if required
+	shouldRotateMonCephxKeys, err := c.mons.RotateMonCephxKeys()
+	if err != nil {
+		return errors.Wrapf(err, "failed to rotate mon cephx keys in the namespace %q", c.ClusterInfo.Namespace)
+	}
+	err = c.mons.UpdateMonCephxStatus(shouldRotateMonCephxKeys)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update cephx status for mon daemons in the namespace %q", c.ClusterInfo.Namespace)
+	}
+
+	// reconcile to restart the mons after cephx key rotation
+	if shouldRotateMonCephxKeys {
+		// reconcile the rook operator so that it will restart the mons after mon cephx key rotation
+		return errors.New(fmt.Sprintf("reconciling the operator to restart mon daemons in the namespace %q after mon cephx key rotation", c.ClusterInfo.Namespace))
```

Comment:
Instead of returning fake error for reconcile, can we not update the mon deployment, to restart the pods ?
Or we are interesting in overall reconcile.